app/components/pay-page/choose-membership-plan-modal/plan-card.ts

```diff
@@ -0,0 +1,73 @@
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import type TimeService from 'codecrafters-frontend/services/time';
+import { formatTimeDurationForCountdown } from 'codecrafters-frontend/utils/time-formatting';
+import type RegionalDiscountModel from 'codecrafters-frontend/models/regional-discount';
+import type PromotionalDiscountModel from 'codecrafters-frontend/models/promotional-discount';
+import type { PricingPlan } from '../choose-membership-plan-modal';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    activeDiscountForYearlyPlan?: PromotionalDiscountModel | null;
+    isSelected: boolean;
+    onPlanSelect: (plan: PricingPlan) => void;
```

Comment:
```suggestion
    onClick: () => void;
```

At the plan-card level, this action is just "onClick". A user can't select a different plan from this card so we can keep the action more simple